src/clipboard_handlers/tables_clipboard.ts

```diff
@@ -69,10 +70,17 @@ export class TableClipboardHandler extends AbstractCellClipboardHandler<
           zones.some((z) => isZoneInside(tableZone, z))
         ) {
           copiedTablesIds.add(table.id);
+          let { numberOfCols, numberOfRows } = zoneToDimension(tableZone);
+          for (let r = tableZone.top; r <= tableZone.bottom; r++) {
+            if (!isCutOperation && !rowsIndexes.includes(r)) {
+              numberOfRows--;
+            }
+          }
```

Comment:
Could you instead just reduce (a copy of) `coreTable.range` here ? instead of adding this new `dimension` ?